src/pages/AddItem.js

```diff
@@ -1,10 +1,30 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { db } from '../lib/firebase';
+import swal from 'sweetalert';
+import { useHistory } from 'react-router-dom';
 
 export default function AddItem(props) {
   const [itemName, setItemName] = useState('');
   const [purchaseFrequency, setPurchaseFrequency] = useState(null);
   const [lastPurchased, setLastPurchased] = useState(null);
```

Comment:
Since items will always be initialized with a lastPurchased value of `null`, storing it in state might be unnecessary.

Reply:
I see what you're saying, but I'm not entirely sure how to get it from the onChange to the `createListItem` without using `useState`. That's really the only pattern I've seen/used until now.

Reply:
Hmm is it supposed to be updated in an onChange? Right now, it looks like `setLastPurchased` isn't being updated except when the values are reset (line 76). If it's not currently set up to have `lastPurchased` as anything other than `null` (which I don't think it will since all new items shouldn't have a last purchased date), you can just include that as a static value in line 67 in `createListItem`: `last_purchased: null`